webapi/src/main/scala/org/knora/webapi/util/StringFormatter.scala

```diff
@@ -1544,7 +1562,7 @@ class StringFormatter private(val maybeSettings: Option[SettingsImpl], initForTe
       * @param iri the IRI to be checked.
       */
     def isKnoraDataIriStr(iri: IRI): Boolean = {
-        DataIriStarts.exists(startStr => iri.startsWith(startStr))
+        isIri(iri) && DataIriStarts.exists(startStr => iri.startsWith(startStr))
```

Comment:
The method `isKnoraDataIriStr` is called in the constructor of `SmartIriImpl`, where `validateAndEscapeIri` has already been called, so we already know it's a valid IRI. Could you refactor this to avoid the redundant validation?

Reply:
done